lib/teleterm/apiserver/handler/handler_cluster_events.go

```diff
@@ -24,6 +26,18 @@ func (s *Handler) ClusterEvents(_ *api.ClusterEventsRequest, stream api.Terminal
 	// TODO: Move the logic from this handler to a separate place.
 	log := s.DaemonService.Log().WithField(trace.Component, "conn:cevents")
 
+	select {
+	case <-time.After(time.Second):
+		message := "Couldn't acquire lock within a second, another stream is already active."
+		log.Info(message)
+		return trace.AlreadyExists(message)
```

Comment:
I don't need to actually wait for a second, I could technically just bail out as soon as I discover that the lock couldn't be acquired.

I was considering using `sync.Mutex.TryLock` for this but its documentation made me research cases where it's actually useful and it didn't seem like the best fit here.

Would this work though?

```go
select {
  case <- s.DaemonService.ClusterEventsSem <- struct{}{}:
    // …
  default:
    return trace.AlreadyExists("Couldn't acquire lock")
}
```

Reply:
Would it work to use an atomic integer as a boolean to specify whether a goroutine is active? Or a `sync/atomic.Bool` if we can use Go 1.19? Would make the logic slightly easier to reason about imo

Reply:
> I think that would work, I don't see why you'd need to wait for a second and try to send on the channel multiple times.

This was purely because of my inexperience with channels. I forgot that I could use `default` here and I remembered it just as I was typing my comment. 😅

Though as Tiago said on Slack, `sync.Mutex.TryLock` would be perfectly good here and as evidenced by the discussion under my post, probably more clear than any other approach.